### PR TITLE
Pull request for redis-lua

### PR DESCRIPTION
**Pull request by jd**
redis: use a Lua script to get measures
redis: create metrics using a single hsetnx() operation
redis: batch new measures retrieval with a Lua script
redis: use a Lua script to list split keys

### CODE REVIEW

**Comment by chungg** [comment]
is there a reason for this? trying to figure out if this fixes something that is worth making the code less readable.

**Comment by jd** [comment]
Performance improvement :)

**Comment by chungg** [comment]
got some benchmarks? :)

**Comment by jd** [comment]
It improves latency and the number of calls to `sendall()/recv()` on the socket is divided by 2, which is a small win. :)

File: gnocchi/storage/redis.py [hunk 1]
```diff
@@ -27,9 +27,36 @@ class RedisStorage(storage.StorageDriver):
...
+local metric_key = KEYS[1]
+local ids = {}
+local cursor = 0
+local substring = "([^" .. ARGV[2] .. "]*)"
+repeat
+    local result = redis.call("HSCAN", metric_key, cursor, "MATCH", ARGV[1])
+    cursor = tonumber(result[1])
+    for i, v in ipairs(result[2]) do
+        -- Only return keys, not values
+        if i % 2 ~= 0 then
```
Comment (chungg) [inline gnocchi/storage/redis.py]: i've no idea what this is for but ok?

Comment (jd) [reply gnocchi/storage/redis.py]: HSCAN returns a table in format `{k1, v1, k2, v2, k3, v3…}` so using this condition allows to ignore `k1, k2, k3` and just push `v1, v2, v3` in `ids`

Comment (chungg) [reply gnocchi/storage/redis.py]: you mean ignore v1, v2, v3...

Comment (jd) [reply gnocchi/storage/redis.py]: Sorry yeah, we want keys here, not values. :)

Comment (chungg) [reply gnocchi/storage/redis.py]: cool cool. yeah i google'd most of the code. seems legit to someone that knows zero lua.

**Comment by jd** [comment]
Rebased.

**Pull request merged**